src/js/actions/tools.js

```diff
@@ -327,13 +332,19 @@ define(function (require, exports) {
                 return currentTool.deselectHandler.call(this, currentTool);
             })
             .then(function () {
-                var psToolName = nextTool.nativeToolName,
+                var psToolName = nextTool.nativeToolName.call(this),
                     setToolPlayObject = toolLib.setTool(psToolName);
 
                 // Set the new native tool
                 return descriptor.playObject(setToolPlayObject);
             })
             .then(function () {
+                if (!nextTool.handleVectorMaskMode) {
```

Comment:
Not a fan of this, this code shouldn't have any tool specific logic here. Why not include the dispatch in the select handler of whatever tool we're being sent to, or the deselect handler of the vector edit tool?